lib/elasticsearch/client_extensions.rb

```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+module Elasticsearch
+  module ClientExtensions
+    def verify_elasticsearch
```

Comment:
Are we ok ignoring the few checks at the beginning of the method? https://github.com/elastic/elasticsearch-ruby/blob/v7.17.10/elasticsearch/lib/elasticsearch.rb#L49-L68

I assume they would not take this PR, but it'd be great if the headers check on this line - https://github.com/elastic/elasticsearch-ruby/blob/v7.17.10/elasticsearch/lib/elasticsearch.rb#L74 - could be pulled to one method and we could just patch that but leave the rest intact.

Reply:
All of those checks are pretty much Elasticsearch-specific, OpenSearch has a different versioning scheme, different name, and different tagline.

Reply:
Sounds good -- might be good to add a spec here to the ES coverage showing that when the server returns some "invalid" header/response value, we just ignore it and don't error.

Reply:
It's a bit hackish because OpenSearch and Elasticsearch containers expect different environment variables (and OpenSearch will fail to start if given Elasticsearch's environment variables), but I managed to add it to the matrix!

Reply:
Ah, even better idea, nice.

Given the spec/search coverage ... would the existing specs there have failed when we version bumped the gem (prior to introduction of this monkey patch)? I'm guessing yes because I assume the verification runs on any connection at all, but want to confirm.

Reply:
I think so, although I haven't actually tried that on the CI.